### DISCUSSION TITLE
Parallel queries from two components

### DISCUSSION THREAD

**Discussion by lishine** [TanStack/query#361] [General] [answered]
Is it a bug or a feature?
In one component
```
const { status, isFetching } = useQuery('gallery',[variables], fetchGallery, { onSuccess: () => {...} })
```
In another component
```
  const { data: rows = [], ...rest } = useQuery('gallery', undefined, { manual: true })
```

The result: both queries return loading status, do retries and eventually error.
If I remove the second one, first one is loading as it should be.

I try to use here the cache from one component in another

**Comment by tannerlinsley** [comment]
This is intentional. My suggestion:
- Extract your gallery query into a custom hook that can accept optional config. This way its reusable, and you'll never find yourself without the query function for that query.
- Extract your function to a different file that you can import where you use this query
- Use `queryCache.getQueryData` to synchronously access the query's data, which doesn't require a fetcher function to be passed at all.

**Comment by lishine** [comment]
I understand the recommended approach about query extraction. Thanks.
`queryCache.getQueryData` application is not clear to me because it will not re-render the component, this is why I useQuery wherever I need this data.

**Comment by tannerlinsley** [comment, marked as answer]
Right. If you need to *subscribe* to the query, you'll have to use `useQuery`. I recommend a custom hook. I do this almost all the time.

**Comment by lishine** [comment]
Github new interface :)

**Comment by lishine** [comment]
May I suggest some contribution to documentation

**Reply by tannerlinsley** [reply]
Absolute. Feel free to create a PR with your suggestions and we'll review them there.
